tuna/fin_class.py

```diff
@@ -211,21 +208,44 @@ def applicability(self):
 
     return True
 
-  def set_all_configs(self):
+  def set_all_configs(self, idx=0, num_blk=1):
     """Gathering all configs from Tuna DB to set up fin input file"""
     with DbSession() as session:
       query = session.query(
           self.dbt.config_table).filter(self.dbt.config_table.valid == 1)
 
       if self.label:
-        query_cfgs = session.query(self.dbt.job_table)\
-            .filter(self.dbt.job_table.reason == self.label)
+        query_cfgs = session.query(self.dbt.config_tags_table)\
+            .filter(self.dbt.config_tags_table.tag == self.label)
         rows = query_cfgs.all()
         ids = tuple([str(job_row.config) for job_row in rows])
         query = query.filter(self.dbt.config_table.id.in_(ids))
 
+      query = query.order_by(self.dbt.config_table.id.asc())
       rows = query.all()
-      for row in rows:
+
+      extra = len(rows) % num_blk
+      start = idx * (len(rows) // num_blk)
+      end = (idx + 1) * (len(rows) // num_blk)
+      if idx < extra:
+        start += idx
+        end += 1 + idx
+      else:
+        start += extra
+        end += extra
```

Comment:
Can you explain what you are doing here? Maybe add some comments to the code

Reply:
This is splitting out the work into blocks for each process. So each process gets a non-overlapping set of configurations to test.

Reply:
OK, it would be useful to have this as a code comment.

Reply:
This can also be accomplished using `limit` and `offset` in MySQL however, I refrained from suggesting that since that way would the MySQL session would be open much longer.